File: micro-service-weave/src/Services/Identity/Identity.Domain/Entities/Menu.cs
using Shared.Domain.Attributes;

namespace Identity.Domain.Entities;

[Table("Identidad", "Menu", "MenuId")]
public class Menu
{
    public long MenuId { get; set; }
    public long? MenuPadreId { get; set; }
    public string Titulo { get; set; } = string.Empty;
    public string? Ruta { get; set; }
    public string? TipoIcono { get; set; }
    public string? Icono { get; set; }
    public string? Clase { get; set; }
    public bool? EsTituloGrupo { get; set; }
    public string? Badge { get; set; }
    public string? BadgeClase { get; set; }
}

File: react-project/src/models/Menu.ts
export interface Menu {
  menuId: number;
  menuPadreId?: number;
  titulo: string;
  ruta?: string;
  tipoIcono?: string;
  icono?: string;
  clase?: string;
  esTituloGrupo: boolean;
  badge?: string;
  badgeClass?: string;
  children?: Menu[];
}

export interface MenuData {
  menus: Menu[];
}

export interface MenuTreeNode extends Menu {
  level: number;
  isExpanded: boolean;
  hasChildren: boolean;
} 